ros2_batch_job/packaging.py

```diff
@@ -24,24 +24,19 @@
 
 
 def build_and_test_and_package(args, job):
-    print('# BEGIN SUBSECTION: build')
-    # ignore ROS 1 bridge package for now
-    ros1_bridge_path = os.path.join(args.sourcespace, 'ros2', 'ros1_bridge')
-    info('ROS1 bridge path: %s' % ros1_bridge_path)
-    ros1_bridge_ignore_marker = None
-    if os.path.exists(ros1_bridge_path):
-        ros1_bridge_ignore_marker = os.path.join(ros1_bridge_path, 'COLCON_IGNORE')
-        info('ROS1 bridge path ignore file: %s' % ros1_bridge_ignore_marker)
-        with open(ros1_bridge_ignore_marker, 'w'):
-            pass
-
-    # Now run build
+    print('# BEGIN SUBSECTION: build underlay packages')
+    if args.mixed_ros_overlay_pkgs:
+        print('Skipping the following packages from underlay:')
+        for skip_pkg in sorted(args.mixed_ros_overlay_pkgs):
+            print('-', skip_pkg)
+
     cmd = [
         args.colcon_script, 'build',
         '--base-paths', '"%s"' % args.sourcespace,
         '--build-base', '"%s"' % args.buildspace,
         '--install-base', '"%s"' % args.installspace,
     ] + (['--merge-install'] if not args.isolated else []) + \
+        (['--packages-skip', ' '.join(args.mixed_ros_overlay_pkgs)]) + \
```

Comment:
This is still wrong. Each argument needs to passed separately - not a single argument with spaces.

Reply:
Are you sure? Looking at the current test build it seems to pass:

```
17:35:24 ==> . "/opt/ros/melodic/setup.sh" && . "/home/jenkins-agent/workspace/linux_packaging_post_rosbridge/ws/rti.sh" && /home/jenkins-agent/workspace/linux_packaging_post_rosbridge/venv/bin/colcon build --base-paths "src" --build-base "build" --install-base "install" --merge-install --packages-skip ros1_bridge ros1_rosbag_storage_vendor rosbag2_bag_v2_plugins --event-handlers console_cohesion+ console_package_list+ --cmake-args -DBUILD_TESTING=OFF --no-warn-unused-cli -DCMAKE_BUILD_TYPE=RelWithDebInfo -DINSTALL_EXAMPLES=OFF -DSECURITY=ON
17:35:24 ###############################################################################
17:35:24 
17:35:24          (c) Copyright, Real-Time Innovations, All rights reserved.            
17:35:24                                                                                
17:35:24                            RTI Connext DDS 5.3.1                       
17:35:24 
17:35:24 ###############################################################################
17:35:28 Topological order
17:35:28 - ament_cppcheck (ros.ament_python)
17:35:28 - ament_flake8 (ros.ament_python)
17:35:28 - ament_package (ros.ament_python)
17:35:28 - ament_pep8 (ros.ament_python)
[...]
17:35:28 - rosbag2_tests (ros.ament_cmake)
17:35:28 - rviz_default_plugins (ros.ament_cmake)
17:35:28 - rviz2 (ros.ament_cmake)
```

Neither the `ros1_bridge` nor the `rosbag2_bag_v2_plugins` are built.

https://ci.ros2.org/view/All/job/linux_packaging_post_rosbridge/39/consoleFull